rust/datafusion/src/execution/context.rs

```diff
@@ -375,15 +380,39 @@ impl ScalarFunctionRegistry for ExecutionContext {
     }
 }
 
+/// Provides OptimizerRule instances to
+///
+/// Because OptimizerRule's themselves need
+/// to be mutable to conform to the OptimizerRuleTrait, they
+/// must be instantiated every time a plan is optimized
+/// ExecutionContext::optimize doesn't have a mutable reference to
+/// &self....
+pub trait OptimizerRuleSource {
+    /// Return the OptimizerRules to apply to a LogicalPlan. The rules
+    /// are applied in the order they are returned in the Vec.
+    fn rules(&self) -> Vec<Box<dyn OptimizerRule>>;
+}
+
+/// Supplies no additional optimizer rules
+struct DefaultOptimizerRuleSource {}
+
+impl OptimizerRuleSource for DefaultOptimizerRuleSource {
+    fn rules(&self) -> Vec<Box<dyn OptimizerRule>> {
+        vec![]
+    }
+}
+
 /// Configuration options for execution context
 #[derive(Clone)]
 pub struct ExecutionConfig {
     /// Number of concurrent threads for query execution.
     pub concurrency: usize,
     /// Default batch size when reading data sources
     pub batch_size: usize,
-    /// Optional physical planner to override the default physical planner
-    physical_planner: Option<Arc<dyn PhysicalPlanner>>,
+    /// Physical planner for converting from `LogicalPlan` to `ExecutionPlan`
+    physical_planner: Arc<dyn PhysicalPlanner>,
```

Comment:
I thought the code would be neater if there was always a physical_planner (which is an instance of the `DefaultPhysicalPlanner`)